pkg/apiclient/argo-kube-client.go

```diff
@@ -86,21 +110,59 @@ func newArgoKubeClient(ctx context.Context, clientConfig clientcmd.ClientConfig,
 	if err != nil {
 		return nil, nil, err
 	}
-	return ctx, &argoKubeClient{instanceIDService, wfClient}, nil
+
+	client := &argoKubeClient{
+		opts:              opts,
+		instanceIDService: instanceIDService,
+		wfClient:          wfClient,
+	}
+	err = client.startStores(restConfig, namespace)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return ctx, client, nil
+}
+
+func (a *argoKubeClient) startStores(restConfig *restclient.Config, namespace string) error {
+	if a.opts.UseCaching {
```

Comment:
This was the intention - not to introduce breaking change. In the same time my team is using this exact method and it would make tremendous difference to be able to have caching in in `argoKubeClient`. The code that depends on this is tested - it's basically server code. However you are right - I didn't add tests for setting it in this place. I'll fix it.

Reply:
I'm not sure why you consider the caching version a breaking change? What does it break?

This PR is marked as a performance improvement, but doesn't improve the performance of the product, only of your usage of it as a go-client? Why wouldn't everyone want this enabled? It uses more memory...

Reply:
> This PR is marked as a performance improvement, but doesn't improve the performance of the product, only of your usage of it as a go-client? Why wouldn't everyone want this enabled? It uses more memory..

This is enabled by default for argo server and all tests for argo server are using this imrovement. This part disables it for argocli user-facing commands in a case that you are using kubectl connection. So if you use argocli for submitting workflow you won't need to wait for informer to synchronise all templates.

Reply:
@Joibel can you re-review this PR. I did answer your comments - if this is not enough please let me know.